subprojects/core/src/main/java/org/gradle/api/tasks/AbstractExecTask.java

```diff
@@ -38,22 +43,28 @@
  */
 public abstract class AbstractExecTask<T extends AbstractExecTask> extends ConventionTask implements ExecSpec {
     private final Class<T> taskType;
+    private final Property<ExecResult> execResult;
```

Comment:
I'd think about using an abstract getter instead of doing the plumbing manually.

Reply:
I don't know if we can do that because if there are plugins that extend AbstractExecTask or Exec, they'll start to fail to compile. Unit tests that create instances outside of our infrastructure would also fail.  Is that an acceptable breakage?